ci_feature/main.sh

```diff
@@ -76,17 +78,34 @@ if [ -e "/etc/omsagent-secret/WSID" ]; then
       else
             domain="opinsights.azure.com"
       fi
-      echo "Making curl request to oms endpint with domain: $domain"
-      curl --max-time 10 https://$workspaceId.oms.$domain/AgentService.svc/LinuxAgentTopologyRequest
+
+      if [ -e "/etc/omsagent-secret/PROXY" ]; then
+            export PROXY_ENDPOINT=$(cat /etc/omsagent-secret/PROXY)
+      fi
+
+      if [ ! -z "$PROXY_ENDPOINT" ]; then
+         echo "Making curl request to oms endpint with domain: $domain and proxy: $PROXY_ENDPOINT"
+         curl --max-time 10 https://$workspaceId.oms.$domain/AgentService.svc/LinuxAgentTopologyRequest --proxy $PROXY_ENDPOINT
```

Comment:
what will happen here if $proxy_endpoint is not a valid uri ?

Reply:
> what will happen here if $proxy_endpoint is not a valid uri ?
Onboarding is going to fail either not valid uri or proxy server doesnt exit or not reachable.
As I mentioned to your other comment, validation will be done at the HELM client so that customer can  see the error  during the deployment of the chart itself.

Reply:
If so, this should be the case/assumption everywhere. We should not log .fatal from fluentbit due to invalid proxy, but let it thru here.

Reply:
Added validation in main.sh and removed log.fatal* invocations.